src/promptflow/promptflow/core/_flow.py

```diff
@@ -270,6 +289,53 @@ def _load(cls, path: Path, data: dict, **kwargs):
             raise UserErrorException(f"Entry function is not specified for flow {path}")
         return cls(path=path, code=code, entry=entry, data=data, **kwargs)
 
+    # region overrides
+    @classmethod
+    def load(
+        cls,
+        source: Union[str, PathLike],
+        raise_error=True,
+        **kwargs,
+    ) -> "EagerFlow":
+        """
+        Direct load eager flow from YAML file.
+
+        :param source: The local yaml source of a flow. Must be a path to a local file.
+            If the source is a path, it will be open and read.
+            An exception is raised if the file does not exist.
+        :type source: Union[PathLike, str]
+        :param raise_error: Argument for eager flow raise validation error on unknown fields.
+        :type raise_error: bool
+        :return: A EagerFlow object
+        :rtype: EagerFlow
+        """
+        _, flow_path = cls._load_prepare(source)
+        with open(flow_path, "r", encoding=DEFAULT_ENCODING) as f:
+            flow_content = f.read()
+            data = load_yaml_string(flow_content)
+        is_eager_flow = cls._is_eager_flow(data)
+        if not is_eager_flow:
+            raise UserErrorException("Please load an eager flow with EagerFlow.load method.")
+        return cls._load(path=flow_path, data=data, **kwargs)
+
+    def _init_executable(self):
+        from promptflow.contracts.flow import EagerFlow as ExecutableEagerFlow
+
+        # TODO(2991934): support environment variables here
+        meta_dict = generate_flow_meta(
+            flow_directory=self.code,
+            source_path=self.entry_file,
+            entry=self.entry,
+            dump=False,
+        )
+        return ExecutableEagerFlow.deserialize(meta_dict)
+
+    def __call__(self, *args, **kwargs):
+        """Direct call of eager flow WILL cause exceptions."""
+        raise UserErrorException("Eager flow can not be called as a function.")
```

Comment:
Let's use the type name, not write "eager flow" in error message. which is not user facing